src/charm.py

```diff
@@ -30,10 +31,35 @@ def __init__(self, *args):
         super().__init__(*args)
         self.framework.observe(self.on.install, self._on_install)
         self.framework.observe(self.on.config_changed, self._on_config_changed)
+        # relation
+        self.framework.observe(self.on.bind_exporter_relation_joined,
+                               self._on_bind_exporter_relation_changed)
+        self.framework.observe(self.on.bind_exporter_relation_changed,
+                               self._on_bind_exporter_relation_changed)
+        self.framework.observe(self.on.bind_exporter_relation_departed,
+                               self._on_prometheus_relation_departed)
+        # initialise stored data
         self._stored.set_default(listen_port=DEFAULT_LISTEN_PORT,
                                  stats_groups=DEFAULT_STATS_GROUPS,)
+
         self.unit.status = ActiveStatus("Unit is ready")
 
+    def _manage_prometheus_bind_exporter_service(self):
+        """Manage the prometheus-bind-exporter service."""
+        logger.debug("prometheus-bind-exporter configuration in progress")
```

Comment:
Should this debug log list the private_address and listen_port used? It may help troubleshooting.